release/models/network-instance/openconfig-network-instance.yang

```diff
@@ -1204,8 +1215,16 @@ module openconfig-network-instance {
 
     leaf name {
       type string;
+      default "DEFAULT";
       description
-        "A unique name for the protocol instance";
+        "A unique name for the protocol instance.
+
+        If the operator does not designate a name for the protocol
+        instance (e.g. config), the implementation should use the
+        name of 'DEFAULT' (e.g. state).  In addition, for
```

Comment:
So, to understand here -- if we were to have a device come up with no configuration, it should use `DEFAULT` as the value, in the case that a client changes this name to `MY_FANCY_DEFAULT_INSTANCE_NAME` then I expect that there would be deletes for that name, and then readvertisement of all state under that new path. Is that a reasonable expectation?

Reply:
More so for the case of non-OpenConfig modeled configuration (e.g. native) or some domain underneath `network-instances` is implied natively is when the default values would be utilized.

For example - native configuration of BGP in an implementation but a client is looking to consume state data according to OpenConfig modeling.  Since various user-defined keys must be configured to give context to formulate the path, these defaults are essentially "helpers" in addition to guidance/recommendations for consistency should one configure via OpenConfig and an implementation only supports a single instance.

Since these are still user-defined (that part is not changing) then yes a client could change the name or initially provision one to their liking (should the implementation support).  The implementation would then need to migrate and adapt to the new keys which in reality brings up operational considerations of in-flight migrations and expectations of this magnitude (but outside the scope of this PR)

Reply:
SGTM